pkg/controller/core/workload_controller.go

```diff
@@ -249,3 +259,28 @@ func workloadStatus(w *kueue.Workload) string {
 	}
 	return pending
 }
+
+// We do not verify Pod's RuntimeClass legality here as this will be performed in admission controller.
+// As a result, the pod's Overhead is not always correct. E.g. if we set a non-existent runtime class name to
+// `pod.Spec.RuntimeClassName` and we also set the `pod.Spec.Overhead`, in real world, the pod creation will be
+// rejected due to the mismatch with RuntimeClass. However, in the future we assume that they are correct.
+func handlePodOverhead(wl *kueue.Workload, c client.Client) {
+	ctx := context.Background()
+
+	for i, pod := range wl.Spec.PodSets {
+		if pod.Spec.RuntimeClassName != nil && len(pod.Spec.Overhead) == 0 {
+			var runtimeClass nodev1.RuntimeClass
+			if err := c.Get(ctx, types.NamespacedName{Name: *pod.Spec.RuntimeClassName}, &runtimeClass); err != nil {
+				klog.Error(err, "failed to get RuntimeClass")
+				continue
+			}
+			if runtimeClass.Overhead != nil {
+				wl.Spec.PodSets[i].Spec.Overhead = runtimeClass.Overhead.PodFixed
+			}
+		} else if pod.Spec.RuntimeClassName == nil && len(pod.Spec.Overhead) > 0 {
+			// In real world, pod creation will be rejected for setting Pod Overhead without corresponding RuntimeClass,
```

Comment:
Why do this here?

Maybe just leave it as is and let the pod creation fail.

Reply:
We just want to calculate more accurately, pod here is just a copied obj, will not change the origin behavior(reject pod creation).

Reply:
But this shouldn't happen, so we might as well not have code (and tests) for this.

Reply:
Yes, normally, users will not set the `Overhead` but leave the `pod.Spec.RuntimeClassName` empty. I'll modify it as you advised.